quickcheck/Test/Tasty/QuickCheck.hs

```diff
@@ -263,28 +256,12 @@ quickCheck :: (Progress -> IO ())
            -> QC.Args
            -> QC.Property
            -> IO QC.Result
-quickCheck yieldProgress args@(QC.Args {QC.maxSuccess = argMaxSuccess}) prop = do
-  -- Here we rely on the fact that QuickCheck currently prints its progress to
-  -- stderr and the overall status (which we don't need) to stdout
-  maxSuccess <- newIORef $ fromIntegral argMaxSuccess
-  let setMaxTests r = case maybeNumTests r of
-        Just n -> unsafePerformIO $ writeIORef maxSuccess (fromIntegral n) $> r
-        Nothing -> r
-  tm <- QC.newTerminal
-          (const $ pure ())
-          $ \progressText -> do
-            ms <- readIORef maxSuccess
-            yieldProgress emptyProgress { progressPercent = 100 * parseProgress progressText / ms }
-  QC.withState args $ \ s ->
-    QC.test s { QC.terminal = tm } $ mapResult setMaxTests prop
-
-  where
-    -- QuickCheck outputs something like "(15461 tests)\b\b\b\b\b\b\b\b\b\b\b\b\b"
-    parseProgress :: String -> Float
-    parseProgress = maybe 0 (\n -> fromIntegral (n :: Int) / fromIntegral (QC.maxSuccess args))
-                  . readMaybe
-                  . takeWhile Char.isDigit
-                  . drop 1
+quickCheck yieldProgress args
+  = (.) (QC.quickCheckWithResult args)
+  $ QCP.callback 
+  $ QCP.PostTest QCP.NotCounterexample
+  $ \QC.MkState {QC.maxSuccessTests, QC.numSuccessTests} _ ->
+    yieldProgress $ emptyProgress {progressPercent = fromIntegral numSuccessTests / fromIntegral maxSuccessTests}
```

Comment:
Wow. This looks awesome, much better than before.

Reply:
One thing that bothers me is that shrinks and discards are not reported. `callback` makes this data very accessible with [State](https://github.com/nick8325/quickcheck/blob/7e972cadfe9745db47a0e625c20096ec854da7f8/src/Test/QuickCheck/State.hs#L18) and I think these should be reported.

Reply:
It's important to keep progress text short, otherwise TUI is likely to break. I don't think there is space to fit information about discarded / shrinks. 

Let's not scope creep please, I have very limited resource to review things here.